### DISCUSSION TITLE
Right side text alignment for JudgementCounterDisplay

### DISCUSSION THREAD

**Discussion by Schmittma** [ppy/osu#26390] [Ideas]
Currently, the text for the JudgementCounterDisplay (Counter direction vertical) is always aligned to the left. 
This works well when put on the left side of the screen. However, when put on the right side of the screen, it might be preferable to have right side text-alignment.

Here represented by flipping horizontally:
![image](https://github.com/ppy/osu/assets/20574467/9711ff46-89ec-4659-9aa6-557716482c83)

Additionally, left-side alignment creates a somewhat suboptimal visual when the text enlarges.

![osu!_5ZU6v3sDNG](https://github.com/ppy/osu/assets/20574467/43679a15-b54e-4367-9533-4c9ef80aa398)

When an origin other than the left side is selected, this gets even worse due to the entire text moving to the left.

![osu!_Gut91oZpJo](https://github.com/ppy/osu/assets/20574467/372d985e-d86a-4fe0-a57d-af82e7808bcc)

**Comment by peppy** [comment]
@Schmittma to keep things simple, do you think it would be fine to automatically decide the alignment based on the position of the counter on the screen? are there actually cases you can imagine where it would be right-anchored but left aligned?

**Reply by Schmittma** [reply]
That would be a good way to handle it. I can currently not think of any niche cases where this would not be preferred.

The only problem i could see is when people anchor in the Top/Bottom-middle.

**Reply by AllWillFreeze** [reply]
Sorry for the necro. The behaviour from HUD elements are bugging me to no end.

Personally, I would love to see a selectable anchor point. Automated could work in most cases, but as mentioned by Scmittma, top/bottom alignment could cause edge cases.

This alignment issue is not limited to a few of the actively changing elements. LegacyAccuracyCounter, TrianglesPerformancePointsCounter, ArgonPerformancePointsCounter also displayed these issues. I'm not entirely sure the full extent of it, but I can assume anything that needs to increase the size of it's display box when a value larger then defaul is added, such as 2 numbers > 3, 6 numbers > 7, etc etc.

Essentially the anchorpoint is defaulting to the centre of the object, and when the box expands, it drives the edges to different areas. I captured both TrianglesPerformancePointsCounter and JudgementCounterDisplay doing it. 

![Ks1s6Ex](https://github.com/ppy/osu/assets/62161627/9533904e-dc49-4b61-b24a-be5b00b1d5eb)

Being able to select the anchor point would solve the issue entirely.

**Reply by n0Oo0Oo0b** [reply]
> are there actually cases you can imagine where it would be right-anchored but left aligned?

With the option to anchor elements to the playfield border now, I think placing it next to the border is a valid case:

![image](https://github.com/user-attachments/assets/a1b0c3d5-1d36-4f0c-b6ec-106be5e529c0)
(top left anchor, top right origin, should be right aligned)

Using the origin instead should do the trick, although getting a configuration like the above would require manually setting the anchor/origin.
